src/effects/VST3/VST3Effect.cpp

```diff
@@ -1217,3 +1183,128 @@ void VST3Effect::ReloadUserOptions()
 
    SetBlockSize(mUserBlockSize);
 }
+
+
+bool VST3Effect::TransferDataToWindow(const EffectSettings& settings)
+{
+   // Update parameter values in the plugin
+   if (StoreSettings(GetSettings(settings)))
```

Comment:
The component handler operations need to be lifted out of `StoreSettings` and called directly from this function.